Add agentless-scanner

0intro opened the pull request:
### What does this PR do?

This change adds the `agentless-scanner` software.

Development branch. Not for review.

### Motivation

<!--
* What inspired you to submit this pull request?
* Link any related GitHub issues or PRs here.
-->

### Additional Notes

<!--
* Anything else we should know when reviewing?
* Include benchmarking information here whenever possible.
* Include info about alternatives that were considered and why the proposed
  version was chosen.
-->

### Possible Drawbacks / Trade-offs

<!--
* What are the possible side-effects or negative impacts of the code change?
-->

### Describe how to test/QA your changes

<!--
* Write here in detail or link to detailed instructions on how this change can
  be tested/QAd/validated, including any environment setup.
-->

### Reviewer's Checklist
<!--
* Authors can use this list as a reference to ensure that there are no problems
  during the review but the signing off is to be done by the reviewer(s).

Note: Adding GitHub labels is only possible for contributors with write access.
-->

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

zhuminyi committed:
bug fix tlshandshakeerrorwriter log level

jinroh committed:
[agentless-scanner] Logs: cleanup and always prefix with scan task when possible

0intro committed:
[agentless-scanner] Bump Trivy version to upstream 2023-12-19

This change bumps Trivy version to upstream 2023-12-19.

It also includes 2 patches:
 - allow restricting filesystem walk to specific folders (onlyDirs)
 - use same walker for slow and fast modes

See lebauce/bump-trivy-upstream branch, which relies on
https://github.com/DataDog/trivy/commits/lebauce/new-main-dd.

jinroh committed:
[agentless-scanner] Types: embed scannerOptions in scanResult

0intro committed:
[agentless-scanner] inv tidy-all

0intro committed:
[agentless-scanner] go.mod 1.20

jinroh committed:
[agentless-scanner] cleanup: dedicated context for umounting containers

jinroh committed:
[agentless-scanner] limits: reduce default AWS limits

jinroh committed:
[agentless-scanner] Rely on start time of scanner for stats

0intro committed:
[trivy] Fix build of image_util_test after Trivy bump

jinroh committed:
[agentless-scanner] multi-process: forward scanner stdout logs

jinroh committed:
[agentless-scanner] nbd: better probing for non busy device

jinroh committed:
[agentless-scanner] forward context canceled errors

jinroh committed:
[agentless-scanner] Pass down awsLimits to newHTTPClientWithAWSStats

jinroh committed:
[agentless-scanner] CLI: early error is passing wrong --actions

jinroh committed:
[agentless-scanner] cleanup: fix cleanup of volumes that could not be attached

jinroh committed:
[agentless-scanner] attach: fix nextXenDevice

jinroh committed:
[agentless-scanner] attach: start counting from 0 for selecting device name

jinroh committed:
[agentless-scanner] Logs: prefix with scan task when possible in scanners

jinroh committed:
[agentless-scanner] Add a stop() method on scanner

jinroh committed:
[agentless-scanner] Add a pushConfig helper on the scanner

safchain committed:
[backport][CWS] fixes of ptrace on ARM (#22240)

* [CWS] fix clone/exec on arm (#22191)

[CWS] fix clone/exec on arm

(cherry picked from commit af39a8ae2bbe9870e084547548e8eb6fd60b2440)

* [CWS] fix ptrace cont on ARM (#22229)

[CWS] fix ptrace cont on ARM

(cherry picked from commit 722d87d3f7c66f36c0fcf2de909e54ea483fe75c)

guyarb committed:
usm: Fix cleanup after startup failure (#22209) (#22215)

usm: Fix cleanup after startup failure

guyarb committed:
usm: classification: kafka: Add a missing configuration (#22232) (#22236)

kacper-murzyn committed:
Update release.json and Go modules for 6/7.51.0-rc.4 (#22225)

[release] Update release.json and Go modules for 6/7.51.0-rc.4

jinroh committed:
[agentless-scanner] Factor out attach-volume detach/cleanup

jinroh committed:
[agentless-scanner] cleanup: handle some corner cases for clean-slate

jinroh committed:
[agentless-scanner] containers: fix skipping docker containers with unsupported driver

jinroh committed:
[agentless-scanner] linting

jinroh committed:
[agentless-scanner] containers: fix omitted ImageManifest in docker

yshapiro-57 committed:
Backport UUID fix to 7.51 (#22239)

0intro committed:
[go.mod] Stick k8s.io to 0.27.6

0intro committed:
[agentless-scanner] Don't build the agentless-scanner on heroku

datadog-heroku-agent deb size increase is too large:
  New package size is 277.95MB
  Stable package (7.49.0) size is 166.86MB
  Diff is 111.09MB > 80.00MB (max allowed diff)

Commit:
[omnibus] Pin oracledb cryptography dependency (#22268) (#22272)

[omnibus] Pin oracledb cryptography dependency (#22268)

jinroh committed:
[agentless-scanner] CLI: add filters for offline mode

0intro committed:
[agentless-scanner] Build agentless-scanner standalone package

This change adds a new package "datadog-agentless-scanner",
which provides the "agentless-scanner" executable and init
scripts. This package depends on the "datadog-agent" package.

0intro committed:
[agentless-scanner] Remove kitchen_centos_*_agent-a6 tests

Remove the kitchen_centos_*_agent-a6 tests, because
the datadog-agent package doesn't fit anymore on the centos
disk image, since the bump of Trivy.

 - kitchen_centos_step_by_step_agent-a6
   At least 53MB more space needed on the / filesystem.

 - kitchen_centos_fips_step_by_step_agent-a6
   At least 54MB more space needed on the / filesystem.

 - kitchen_centos_fips_install_script_agent-a6
   At least 74MB more space needed on the / filesystem.

jinroh committed:
[agentless-scanner] containers: append the correct BOM properties for images to be properly ingested

jinroh committed:
[agentless-scanner] containers: make sure docker image refs are always populated

jinroh committed:
[agentless-scanner] containers: move container tagging inline

jinroh committed:
[agentless-scanner] CLI: fix human parsing for directories

jinroh committed:
[agentless-scanner] containers: lint

jinroh committed:
[agentless-scanner] nbd: avoid crashing in waitServerClosed if server was not properly setup

jinroh committed:
[agentless-scanner] containers: also scan for app vulns

jinroh committed:
[agentless-scanner] CLI: fix empty --filters

jinroh committed:
[agentless-scanner] scanners: small renamings and disable app scan by default for containers

jinroh committed:
[agentless-scanner] containers: add some metrics

jinroh committed:
[agentless-scanner] containers: fix mounting when mount options is greater than pagesize

jinroh committed:
[agentless-scanner] containers: fix image name parsing for docker when image not from repo

kacper-murzyn committed:
Update release.json and Go modules for 6/7.51.0-rc.5 (#22367)

jinroh committed:
[agentless-scanner] AWS: add a DatadogAgentlessScannerHostOrigin tag containing the scanner hostname

KevinFairise2 committed:
Update AMI to use our copies (#22393)

jinroh committed:
[agentless-scanner] containers: various fixes

jinroh committed:
[agentless-scanner] cleanups

jinroh committed:
[agentless-scanner] missing error wrapping

jinroh committed:
[agentless-scanner] lint

jinroh committed:
[agentless-scanner] attach: exclude /dev/xvddy and /dev/xvddz

jinroh committed:
[agentless-scanner] AWS: make sure we always report an account in stats

jinroh committed:
[agentless-scanner] lint

jinroh committed:
[agentless-scanner] AWS: reduce #request for DeleteVolume

jinroh committed:
[agentless-scanner] AWS: get rid of using context to pass limiter/waiter around

jinroh committed:
[agentless-scanner] containers: s/ctrd/containerd/

jinroh committed:
[agentless-scanner] errors: fix some error wrapping

jinroh committed:
[agentless-scanner] lsblk: pass context

jinroh committed:
[agentless-scanner] cleanup: poll block device instead of polling DeleteVolume

jinroh committed:
[agentless-scanner] Reduce parallel number of scans

jinroh committed:
[agentless-scanner] logs: remove useless logs

jinroh committed:
[agentless-scanner] scanners: avoid starvation with a dedicated pool per worker

jinroh committed:
[agentless-scanner] scanner: add a CreatedAt field for scan restuls

jinroh committed:
[agentless-scanner] CLI: add a snapshot command

0intro committed:
[agentless-scanner] Increase datadog-agent allowed pkg size to 100 MB on arm64

check_pkg_size-arm64-a6
datadog-agent deb size increase is too large:
  New package size is 441.02MB
  Stable package (6.49.0) size is 347.12MB
  Diff is 93.90MB > 80.00MB (max allowed diff)
datadog-agent rpm size increase is too large:
  New package size is 448.14MB
  Stable package (6.49.0) size is 354.94MB
  Diff is 93.19MB > 80.00MB (max allowed diff)

check_pkg_size-arm64-a7
datadog-agent deb size increase is too large:
  New package size is 403.94MB
  Stable package (7.49.0) size is 309.03MB
  Diff is 94.91MB > 80.00MB (max allowed diff)
datadog-agent rpm size increase is too large:
  New package size is 409.20MB
  Stable package (7.49.0) size is 314.93MB
  Diff is 94.27MB > 80.00MB (max allowed diff)

0intro committed:
[agentless-scanner] Remove kitchen_ubuntu_install_script_(iot_agent|dogstatsd)-a7 tests

These tests are failing since 2023-01-24 for unknown reason.

"body": "{\"error\":{\"code\":\"InvalidTemplateDeployment\",\"message\":\"The template deployment 'deploy-372571a54f3e2b92' is not valid according to the validation procedure. The tracking id is '053f44af-15a9-4b83-b766-7cb32aa96d9e'. See inner errors for details.\",\"details\":[{\"code\":\"InvalidParameter\",\"target\":\"imageReference\",\"message\":\"The following list of images referenced from the deployment template are not found: Publisher: Canonical, Offer: 0001-com-ubuntu-server-focal, Sku: 20_04-lts, Version: 20.04.202004230. Please refer to https://docs.microsoft.com/en-us/azure/virtual-machines/windows/cli-ps-findimage for instructions on finding available images.\"}]}}"

jinroh committed:
[agentless-scanner] cleanup: fix bad err handling in cleanup

jinroh committed:
[agentless-scanner] typo

jinroh committed:
[agentless-scanner] CLI: fix parsing "/"

0intro committed:
[agentless-scanner] Add /usr/lib/sysimage to the list of OnlyDirs

0intro committed:
[agentless-scanner] Remove kitchen_centos_install_script_agent-a6 test

Remove the kitchen_centos_install_script_agent-a6 test,
because the datadog-agent package doesn't fit anymore on
the centos disk image, since the bump of Trivy.

- kitchen_centos_install_script_agent-a6
  At least 74MB more space needed on the / filesystem.

jinroh committed:
[agentless-scanner] nbd: fix race when closing an errored nbd server

jinroh committed:
[agentless-scanner] track a little TODO

0intro committed:
[agentless-scanner] Use a version of Trivy that doesn't depend on trivy-iac

Removing the dependency on trivy-iac makes the agentless-scanner
executable 60 MB smaller. Also, it no longer spend 1s on package
initialization.

0intro committed:
[agentless-scanner] Import modernc.org/sqlite driver

0intro committed:
[agentless-scanner] Fix implementation of onlyDirs in Trivy's shouldSkipFile function

Fix onlyDirs in shouldSkipFile so matching works as expected

This change includes two fixes:
 - Move the "Skipping file" debug print to the right place.
 - Pass the filePath instead of the base directory to the Match
   function since Match("usr/lib/*", "usr/lib") would return false.

L3n41c committed:
Update `datadog-vault-secrets` in our internal images (#22483)

Backport #22482 to 7.51.x: Update `datadog-vault-secrets` in our internal images

0intro committed:
[agentless-scanner] Replace /etc/** by /etc/* in onlyDirs

0intro committed:
[agentless-scanner] Add /usr/lib/* to onlyDirs

The /etc/os-release file is often a symlink to /usr/lib/os-release.

0intro committed:
[agentless-scanner] Install agent + agentless-scanner packages in userdata.sh

chouquette commented on .gitlab/pkg_metrics.yml:
```diff
@@ -303,6 +303,8 @@ check_pkg_size-amd64-a7:
     - dogstatsd_suse-x64
     - iot_agent_suse-x64
+    - agentless_scanner_deb-x64
+    - agentless_scanner_rpm-x64
```
I suppose we could also monitor the suse package here

0intro replied:
Right. I've added the SUSE package as well.

chouquette commented on .gitlab/trigger_release.yml:
```diff
@@ -20,7 +20,7 @@
     - source /root/.bashrc
     - export RELEASE_VERSION=$(inv agent.version --major-version 7 --url-safe --omnibus-format)-1
-    - inv pipeline.trigger-child-pipeline --no-follow --project-name "DataDog/agent-release-management" --git-ref "stable-agent-7.50.x" --variables "ACTION,AUTO_RELEASE,BUILD_PIPELINE_ID,RELEASE_PRODUCT,RELEASE_VERSION,TARGET_REPO,TARGET_REPO_BRANCH"
+    - inv pipeline.trigger-child-pipeline --no-follow --project-name "DataDog/agent-release-management" --git-ref "ducolombier/agentless-scanner" --variables "ACTION,AUTO_RELEASE,BUILD_PIPELINE_ID,RELEASE_PRODUCT,RELEASE_VERSION,TARGET_REPO,TARGET_REPO_BRANCH"
```
just opening a thread to be sure this doesn't get merged as such

chouquette commented on omnibus/config/projects/agentless-scanner.rb:
```diff
@@ -0,0 +1,184 @@
+    runtime_script_dependency :pre, "coreutils"
+    runtime_script_dependency :pre, "grep"
+    if redhat_target?
+      runtime_script_dependency :pre, "glibc-common"
```
as a go only binary, does it actually depend on glibc?

chouquette replied:
ah, good point

chouquette commented on omnibus/config/projects/agentless-scanner.rb:
```diff
@@ -0,0 +1,184 @@
+  parameters({
+    'InstallDir' => install_dir,
+    'InstallFiles' => "#{Omnibus::Config.source_dir()}/datadog-agent/dd-agent/packaging/datadog-agent/win32/install_files",
+    'BinFiles' => "#{Omnibus::Config.source_dir()}/datadog-agent/src/github.com/DataDog/datadog-agent/bin/agent",
```
Shouldn't `bin/agent` be changed?

chouquette commented on omnibus/config/projects/agentless-scanner.rb:
```diff
@@ -0,0 +1,184 @@
+  })
+end
+# OSX .pkg specific flags
+package :pkg do
+  identifier 'com.datadoghq.agentless-scanner'
+  #signing_identity 'Developer ID Installer: Datadog, Inc. (JKFCB4CN7C)'
+end
+compress :dmg do
+  window_bounds '200, 200, 750, 600'
+  pkg_position '10, 10'
+end
```
is macOS supported by this product?

chouquette commented on omnibus/config/templates/datadog-agentless-scanner/sysvinit_debian.agentless-scanner.erb:
```diff
@@ -0,0 +1,163 @@
+AGENTPATH="$INSTALL_DIR/bin/agentless-scanner"
+PIDFILE="$INSTALL_DIR/run/agentless-scanner.pid"
+AGENT_ARGS="--config=$ETC_DIR/datadog.yaml --pidfile=$PIDFILE"
+AGENT_USER="root"
```
Shouldn't this program use the datadog-agent user?

chouquette commented on omnibus/config/templates/datadog-agentless-scanner/sysvinit_debian.agentless-scanner.erb:
```diff
@@ -0,0 +1,163 @@
+
+
+
+if [ ! -x $AGENTPATH ]; then
```
could you enclose the variable in double quotes to ensure it doesn't break if the install path contains spaces?

chouquette commented on omnibus/package-scripts/agentless-scanner-deb/postrm:
```diff
@@ -0,0 +1,31 @@
+case "$1" in
+    purge)
+        echo "Deleting dd-agent user"
+        deluser dd-agent --quiet
+        echo "Deleting dd-agent group"
+        (getent group dd-agent >/dev/null && delgroup dd-agent --quiet) || true
+        echo "Force-deleting $INSTALL_DIR"
```
Wouldn't this break if we uninstall & purge the agentless-scanner but not the agent?

chouquette commented on omnibus/package-scripts/agentless-scanner-deb/postrm:
```diff
@@ -0,0 +1,31 @@
+        (getent group dd-agent >/dev/null && delgroup dd-agent --quiet) || true
+        echo "Force-deleting $INSTALL_DIR"
+        rm -rf $INSTALL_DIR
+        rm -rf $LOG_DIR
+        rm -rf $CONFIG_DIR
```
same question here

chouquette commented on omnibus/config/projects/agentless-scanner.rb:
```diff
@@ -0,0 +1,184 @@
+    # have to list them, because they'll already be there because of preinst
+    runtime_script_dependency :pre, "coreutils"
+    runtime_script_dependency :pre, "grep"
+    if redhat_target?
```
IIRC you mentionned that the agentless scanner depends on the agent but this doesn't appear to be reflected here. It would also probably simplify the package scripts quite a bit since most folders & user setup could be handled by the agent's package script

chouquette commented on tasks/agentless_scanner.py:
```diff
@@ -16,47 +20,157 @@
-    cmd = 'go build -mod={go_mod} {race_opt} {build_type} -tags "{go_build_tags}" '
-    cmd += '-o {agent_bin} -gcflags="{gcflags}" -ldflags="{ldflags}" {REPO_PATH}/cmd/agentless-scanner'
+    # generate windows resources
+    if sys.platform == 'win32':
+        if arch == "x86":
+            env["GOARCH"] = "386"
```
do we still support windows 32bits?

chouquette commented:
also just double checking, is it intended that the target branch is `7.50.x` and not `main`?

jinroh committed:
[agentless-scanner] split: make a types package to store constants and types

jinroh committed:
[agentless-scanner] split: move parsing and constructor in types/

jinroh committed:
[agentless-scanner] split: move scanners in dedicated package

jinroh committed:
[agentless-scanner] split: move nbd code in dedicated package

jinroh committed:
[agentless-scanner] split: move aws waiter code in dedicated awsutils package

jinroh committed:
[agentless-scanner] split: move aws config in dedicated awsutils package

jinroh committed:
[agentless-scanner] split: move devices related code in dedicated package

jinroh committed:
[agentless-scanner] split: move aws EBS/Lambda setup and cleanup code in dedicated awsutils package

jinroh committed:
[agentless-scanner] split: move runner code in dedicated package (remove noattach mode)

jinroh committed:
[agentless-scanner] split: move AWS commands into a dedicated package awscmd

jinroh committed:
wip

jinroh committed:
[agentless-scanner] split: remove any reference to arn.ARN and provide our own resource identifier struct (still called ARN for now)

jinroh committed:
[agentless-scanner] split: rename ARN to CloudID

jinroh committed:
[agentless-scanner] split: various fixes

jinroh committed:
[agentless-scanner] split: keep generic cleanup login in runner

jinroh committed:
[agentless-scanner] split: rename arn.go to cloud_id.go

jinroh committed:
[agentless-scanner] split: rename trivy_inmemory_cache to vulns_trivy_cache

jinroh committed:
[agentless-scanner] split: missing copyrights

ogaca-dd commented on cmd/agentless-scanner/aws-deploy.sh:
```diff
@@ -0,0 +1,263 @@
+#!/bin/bash
```
Can you update the codeowner for files in `cmd/agentless-scanner`?

0intro replied:
Done. Thanks.

jinroh committed:
[agentless-scanner] split: normalize enum namings

jinroh committed:
[agentless-scanner] Store all created resources in one map

jinroh committed:
[agentless-scanner] Fix snapshot created time when retrying

jinroh committed:
[agentless-scanner] More helpful error message in ParseCloudID

jinroh committed:
[agentless-scanner] split: move EC2CloudID to types

jinroh committed:
[agentless-scanner] split: add missing "run" command at root level

0intro committed:
[agentless-scanner] Print number of results in "scanner finished" log print

0intro committed:
[agentless-scanner] Add missing copyright headers

0intro committed:
[agentless-scanner] Add comment of nbd package in nbd_nolinux.go

0intro committed:
[agentless-scanner] Sort and deduplicate imports in tasks/__init__.py

0intro committed:
[agentless-scanner] Use latest buildkit from 0.11 branch

0intro committed:
[agentless-scanner] Always build ebs_backend.go

jinroh committed:
[agentless-scanner] Fix offline mode workers count

jinroh committed:
[agentless-scanner] Fix offline mode workers count

jinroh committed:
[agentless-scanner] Rename ScanType to TaskType and actually parses it on RC inputs

jinroh committed:
[agentless-scanner] Simplify switch/case (thx copilot...)

jinroh committed:
[agentless-scanner] Add invariant to make sure a task is of matching the proper cloud provider

jinroh committed:
[agentless-scanner] CLI: unify flags struct name

jinroh committed:
[agentless-scanner] types: move DiskMode at the task level

jinroh committed:
[agentless-scanner] types: move DiskMode at the task level

0intro committed:
[agentless-scanner] Use * instead of ** in OnlyDirs

jinroh committed:
[agentless-scanner] split: reactivate no-attach mode

jinroh committed:
[agentless-scanner] CLI: keep awscmd in main package and create tools/

jinroh committed:
[agentless-scanner] containers: activate vulncontainers by default

jinroh committed:
[agentless-scanner] CLI: rename scan-type to task-typ

jinroh committed:
[agentless-scanner] Improve logs

jinroh committed:
[agentless-scanner] Draft a support for scanning AMIs

jinroh committed:
[agentless-scanner] ebs: fix error checking

jinroh committed:
[agentless-scanner] awsutils: do not return an error in GetConfig

0intro committed:
[agentless-scanner] Implement a local scan command

jinroh committed:
[agentless-scanner] types: make CloudID fields be private

jinroh committed:
[agentless-scanner] types: more explicit AsText() method for CloudID

jinroh committed:
[agentless-scanner] missing Println

jinroh committed:
[agentless-scanner] cleanup: fix regionsCleanup delegate role

jinroh committed:
[agentless-scanner] types: proper structure for RolesMapping

jinroh committed:
[agentless-scanner] cleanup: simplify the cleanup data structures

jinroh committed:
[agentless-scanner] roles: better handling of role delegation: always default to nominal role name and fallback if the role could not be assumed

jinroh committed:
[agentless-scanner] CLI: minimal automated cloud provider detection

0intro committed:
[agentless-scanner] Improve implementation of onlyDirs in Trivy's shouldSkipDir function

The shouldSkipDir function now calls doublestar.Match for onlyDirs,
so it doesn't traverse all the subdirectories when not necessary.

jinroh committed:
[agentless-scanner] nbd: pass ScanTask as first parameter and improve logging

jinroh committed:
[agentless-scanner] nbd: try to avoid race condition when closing nbd while starting it

jinroh committed:
[agentless-scanner] CLI: allow cross-account offline mode

jinroh committed:
[agentless-scanner] AWS: missing RoleSessionName is AssumeRole call

jinroh committed:
[agentless-scanner] nbd: add missing prefixes to some logs

jinroh committed:
[agentless-scanner] AMI: add offline mode and rely on CopySnapshot for giving the right permissions

jinroh committed:
[agentless-scanner] types: shorter task identifiers

jinroh committed:
[agentless-scanner] AMI: mutualize some codes

jinroh committed:
[agentless-scanner] CLI: namespace aws cmd functions

guyarb committed:
Revert "[NPM][USM] Move probes on sockfd_lookup_light to USM (#21259)" (#22621)

This reverts commit 18169a18e5e8dacffca8a98e5e081c21607e2aa1.

julien-lebot committed:
Make send on start default to true for the Windows registry integration (#22503) (#22618)

[Backport #22503] Make send on start default to true for the Windows registry integration

0intro committed:
[agentless-scanner] Remove modernc.org/sqlite replacement in go.mod

This should prevent error when running "inv -e lint-licenses".

0intro committed:
[agentless-scanner] Fix Python issues reported by the linter

./tasks/agentless_scanner.py:1:1: F401 'datetime' imported but unused
./tasks/agentless_scanner.py:11:1: F401 '.utils.generate_config' imported but unused
./tasks/agentless_scanner.py:11:1: F401 '.utils.get_git_branch_name' imported but unused
./tasks/agentless_scanner.py:11:1: F401 '.utils.get_git_commit' imported but unused
./tasks/agentless_scanner.py:11:1: F401 '.utils.get_go_version' imported but unused
./tasks/agentless_scanner.py:11:1: F401 '.utils.get_gopath' imported but unused
./tasks/agentless_scanner.py:45:14: F821 undefined name 'filter_incompatible_tags'
./tasks/agentless_scanner.py:57:9: F821 undefined name 'build_messagetable'
./tasks/agentless_scanner.py:58:16: F821 undefined name 'versioninfo_vars'
./tasks/agentless_scanner.py:59:9: F821 undefined name 'build_rc'
./tasks/agentless_scanner.py:104:19: F821 undefined name 'Exit'

jinroh committed:
[agentless-scanner] AWS: waiter also return the polled snapshot resource

0intro committed:
[agentless-scanner] Fix StartNBDBlockDevice definition on nolinux

0intro committed:
[agentless-scanner] Fix Python issues reported by the linter

0intro committed:
[agentless-scanner] Set copyright information and license

0intro committed:
[agentless-scanner] Fix Python issues reported by the linter

0intro committed:
[agentless-scanner] Fix imports on nolinux

jinroh committed:
[agentless-scanner] nbd: auto-shutdown staled clients before starting a new server

jinroh committed:
[agentless-scanner] runner: extract a CleanupScanDir that performs umount and dir cleanup

0intro committed:
[agentless-scanner] Add support for Amazon EC2 M4 in detectCloudProvider

github-actions[bot] committed:
Update release.json and Go modules for 6/7.51.0-rc.6 (#22658)

Co-authored-by: GitHub Actions <41898282+github-actions[bot]@users.noreply.github.com>

jinroh committed:
[agentless-scanner] types: add resource tags for scan tasks and use the scanner as pass-through

jinroh committed:
[agentless-scanner] AWS: always provide a User-Agent and RoleSessionName that link back to us

jinroh committed:
[agentless-scanner] AWS: add a GetIdentity method

jinroh committed:
[agentless-scanner] AMIs: offline mode to only "in-use" AMIs

jinroh committed:
[agentless-scanner] CLI: better probing for aws env (region / account)

jinroh committed:
[agentless-scanner] AWS: fix waiter for cross-region and add DescribeImage support

jinroh committed:
[agentless-scanner] AWS: add resource type host-image and allow scanning for AMI ids directly

jinroh committed:
[agentless-scanner] AWS: cleanup waiter loop when no more subs

jinroh committed:
[agentless-scanner] containers: only one container mount prefix

jinroh committed:
[agentless-scanner] types: avoid passing service name to AWSCloudID

GustavoCaso commented:
I think it would be useful to split this PR into smaller more managable PR so the review process is not as dunting.

We could split the work into adding the agentless-scanner code and another PR with all the necessary changes to build a distribute the agentless-scanner and CI steps. 

What do you think?

GustavoCaso commented:
Also, I believe it would be good to add any entry in https://datadoghq.atlassian.net/wiki/spaces/agent/pages/3074132329/Overview+of+Agents+Binaries+and+shared+features with the overview of what the new binary does 😄

gh123man committed:
Finalize 7.51.0 (#22852)

* Update release JSON and go mods

* Add prelude for 7.51.0 release

jinroh committed:
[agentless-scanner] types: fix default parse of ami-* resources

jinroh committed:
[agentless-scanner] types: introduce ScannerID containing hostname and cloud provider infos

jinroh committed:
[agentless-scanner] types: rename ScanTask#CloudID field to TargetID

jinroh committed:
[agentless-scanner] AMI: upgrade agent-payload and use new SBOMSourceType_HOST_IMAGE for image scanning

jinroh committed:
[agentless-scanner] types: unify ScannerName with ScanAction

jinroh committed:
[agentless-scanner] runner: add support for scanning containers app

jinroh committed:
[agentless-scanner] runner: log error when using disk_mode noattach for unsupported actions

jinroh committed:
[agentless-scanner] types: make sure we dedupe scan actions

jinroh committed:
[agentless-scanner] lint

jinroh committed:
[agentless-scanner] containers: store the full mountpoint as part of the Container struct

jinroh committed:
[agentless-scanner] containers: proper tagging of container app vulns

0intro committed:
[agentless-scanner] Apply changes after merge with 7.51.0 tag

It includes the following changes:
 - API changes to Remote Config.
 - API changes to Components.
 - Exclude agentles-scanner command in lint_fxutil_oneshot_test

0intro committed:
[agentless-scanner] Add cmd/agentless-scanner to .github/CODEOWNERS

0intro committed:
[agentless-scanner] Bump aws-sdk-go-v2

go get -u github.com/aws/aws-sdk-go-v2/...

This is necessary, since we want to use the latest version of the API.
Otherwise, we could have error like "not found, ResolveEndpointV2".

0intro committed:
[agentless-scanner] Initialize JavaDB

This change downloads the Java database (820 MB)
into the /tmp/trivy directory. This is required to be
able to retrieve package information about jar files.

This should prevent the following error:

post analysis error: Unable to initialize the Java DB: Java DB update failed: Java DB client not initialized

0intro committed:
[agentless-scanner] Deduplicate agentless_scanner collection

0intro committed:
[agentless-scanner] inv generate-licenses

0intro commented on .gitlab/binary_build/linux.yml:
```diff
@@ -97,3 +97,35 @@ build_iot_agent-binary_arm64:
     - inv check-go-version
     - inv -e agent.build --flavor iot --major-version 7
+
+build_agentless_scanner-binary_x64:
```
Yes, that's a general issue in the yml files.

0intro committed:
[agentless-scanner] Add agentless_scanner_suse-x64 in pkg_metrics

0intro committed:
[agentles-scanner] Don't delete dd-user after uninstalling agentless-scanner deb package

jinroh committed:
[agentless-scanner] cleanup: make sure we record every accounts that we reach for our cleanup process

jinroh committed:
[agentless-scanner] RC: add client.WithProducts

0intro committed:
[agentless-scanner] Add TargetID in success and failure log prints

jinroh committed:
[agentless-scanner] AWS: cloudformation update with delegate roles

0intro committed:
[agentless-scanner] Use github.com/distribution/reference

The github.com/docker/distribution/reference repository is deprecated.

0intro committed:
[agentless-scanner] Add github.com/distribution/reference copyrights

0intro committed:
[agentless-scanner] Bump Trivy

Use the lebauce/new-main-dd branch, updated, and with the addition
of the "Lazily creates zap logger" change.

jinroh committed:
[agentless-scanner] typos

jinroh committed:
[agentless-scanner] devices: increase polling duration for listing partitions

0intro committed:
[agentless-scanner] Bump Trivy to version 2024-02-28

jinroh committed:
[agentless-scanner] vulns: load java DB thread-safe

0intro committed:
[agentless-scanner] Fix build of Agent after bump of Trivy

0intro committed:
[agentless-scanner] Sync Agent's OnlyDirs with main

0intro committed:
[agentless-scanner] Ignore linter warnings

cmd/agentless-scanner/runner/trivy_sbom_convert.go:56:3: SA1019: cyclonedx.Tool is deprecated: Use Component or Service instead. (staticcheck)
cmd/agentless-scanner/runner/trivy_sbom_convert.go:857:22: SA1019: cyclonedx.Tool is deprecated: Use Component or Service instead. (staticcheck)

0intro committed:
[agentless-scanner] Fix licenses after Trivy bump

jinroh committed:
[agentless-scanner] containers: sbomID is always the same

0intro committed:
[agentless-scanner] Sync with ducolombier/bump-trivy-upstream

0intro committed:
[agentless-scanner] Sync with ducolombier/bump-trivy-upstream

jinroh committed:
[agentless-scanner] runner: revert rc WithProducts changes

jinroh committed:
[agentless-scanner] runner: split main loop in smaller functions

jinroh committed:
[agentless-scanner] runner: move scanning code in a new Worker struct

jinroh committed:
[agentless-scanner] tools: enable nbd module

jinroh committed:
[agentless-scanner] runner: add an HTTP server dispatching scan tasks on workers

jinroh committed:
[agentless-scanner] runner: add ability to inspect running scans

jinroh committed:
[agentless-scanner] types: serializable RolesMapping

jinroh committed:
[agentless-scanner] runner: stop reporters when finished sending results

jinroh committed:
[agentless-scanner] types: ParseRolesMapping bubble up errors

jinroh committed:
[agentless-scanner] lint: copyrights

jinroh committed:
[agentless-scanner] waiter: make sure we support paginated results

jinroh committed:
[agentless-scanner] runner: better error logging for RC init

jinroh committed:
[agentless-scanner] tools: remove aws deployment related tooling (now in terraform module repo)

jinroh committed:
[agentless-scanner] lambda: detect service name from tags

jinroh committed:
[agentless-scanner] containers: use SBOMSourceType_CONTAINER_IMAGE_LAYERS for container app scanning

jinroh committed:
[agentless-scanner] containers: remove TODOs

0intro committed:
[agentless-scanner] Disable more Trivy analyzers

This change disables the following analyzers:

- apk-command
- history-dockerfile
- image-config-secret

These analyzer were added more recently in Trivy.

0intro committed:
Merge tag '7.51.0' into ducolombier/agentless-scanner

7.51.0

0intro committed:
Merge remote-tracking branch 'origin/main' into ducolombier/agentless-scanner

0intro committed:
[agentless-scanner] Apply changes after merge with main branch

It includes the following changes:
 - Use eventplatform.Component
 - API changes to pkg/logs/client/http
 - API changes to pkg/logs/pipeline

0intro committed:
[agentless-scanner] Use custom branch of agent-release-management

Use the "ducolombier/agentless-scanner-7.52" branch, so
the agentless-scanner release will be triggered with Agent >= 7.52.

0intro committed:
[agentless-scanner] Implement ScannerConfig structure

This change adds a new ScannerConfig structure, which contains
the configuration of the agentless-scanner. This structure
is passed to all the functions that requires access to the
agentless-scanner configuration.

Since the ScannerConfig is filled in the main package, there
is no need to import pkg/config anymore in any of the other
packages.

0intro committed:
[agentless-scanner] Remove runWithModules in runCmd

Calling runWithModules shouldn't not be required anymore since
runCmd is now a subcommand of rootCommand.

0intro committed:
[agentless-scanner] Use comp/core/config instead of pkg/config

0intro committed:
[agentless-scanner] Replace eventForwarder global by local variables

This change also removes the runWithModules function,
so the eventplatform.Component interface value could
easily be passed back to caller.

0intro committed:
[agentless-scanner] Move cmd/agentless-scanner packages to pkg/agentless

0intro committed:
[agentless-scanner] Rename awsutils package to awsbackend

jinroh committed:
[agentless-scanner] aws: enforce singleflight in GetConfig

jinroh committed:
[agentless-scanner] scanners: getting rid of our last init()

0intro committed:
[agentless-scanner] Replace statsd global by local variables in cmd/agentless-scanner

0intro committed:
[agentless-scanner] Replace statsd global by local variables in pkg/agentless

0intro committed:
[agentless-scanner] Convert agentless-scanner command to the subcommands hierarchy

0intro committed:
Merge remote-tracking branch 'origin/main' into ducolombier/agentless-scanner

0intro committed:
Merge remote-tracking branch 'origin/main' into ducolombier/agentless-scanner

0intro committed:
[agentless-scanner] Use main branch of agent-release-management

It should work as long as we the tag starts with 7.53 and upper.

jinroh committed:
[agentless-scanner] vulns: add Host value in SBOMPayload
